src/annotator/host.coffee

```diff
@@ -4,6 +4,24 @@ Guest = require('./guest')
 
 module.exports = class Host extends Guest
   constructor: (element, options) ->
+
+    # Some options' values are not JSON-stringifiable (e.g. JavaScript
+    # functions) and will be omitted when the options are JSON-stringified.
+    # Add a JSON-stringifiable option for each of these so that the sidebar can
+    # at least know whether the callback functions were provided or not.
+    if options.services?[0]
+      service = options.services[0]
+      if service.onLoginRequest
+        service.onLoginRequestProvided = true
+      if service.onLogoutRequest
+        service.onLogoutRequestProvided = true
+      if service.onSignupRequest
+        service.onSignupRequestProvided = true
+      if service.onProfileRequest
+        service.onProfileRequestProvided = true
+      if service.onHelpRequest
+        service.onHelpRequestProvided = true
```

Comment:
It mutates it, but only by adding new options not by modifying any existing ones (so I don't think it's going to break anything). Having said that I think it'd be good to refactor the settings object to be an object with methods rather than a raw data object, then we could do this kind of work in its constructor method, and it could be protected from mutation by any other parts of the code.

One thing I was trying to do here was to minimise the differences between settings seen by the sidebar code and settings seen by the annotator code. The `on*Request` callback functions are already visible to the annotator code only and not the sidebar code, that's unavoidable. But I tried to ensure that the new `on*RequestProvided` booleans would be visible to both sides, even though the annotator code doesn't technically need these settings, just to minimise any further branching of the (imo already pretty confusing) settings system.

For the same reason I also added new `on*RequestProvided` settings instead of reusing the existing `on*Request` names to avoid having `on*Request` be a function in the annotator code but the same setting be a boolean in the sidebar code.

(In future I think we could potentially even iron-out the fact that `on*Request` callback functions exist annotator-side only, a sidebar-side settings object could have `on*Request` functions that look exactly the same but that internally send the message across the bridge to the sidebar-side settings object.)